### ISSUE TITLE
Use Hydra multirun mode to do optimization tasks

### ISSUE THREAD

**Issue by ri-heme** [RasmussenLab/MOVE#38]
The hyperparameter optimization can be re-adapted to make use of Hydra's multirun mode (and/or sweeper).

## Overview of tasks:

Below a list of tasks to convert these two modules from its current state:

### Module and schema

- [x] Define new schemas for this task: `OptimizeTaskConfig` (inherits from `TaskConfig`) class in `move.conf.schema`.
- [x] Create new `move.tasks.optimize_hyperparameters` module and function.
- [x] Create example `experiment` config for hyperparameter search. See [example YAML](https://github.com/ri-heme/example-hydra-app/blob/main/config/experiment/multirun.yaml) in [hydra-app-example](https://github.com/ri-heme/example-hydra-app#multirun).

### Function

For the function `optimize_hyperparamters`:

- [x] Log job number 
- [x] Create objective function TSV (if it doesn't exist)
- [x] Load pre-processed data
- [x] Split data into training/test sets
- [x] Make dataloaders
- [x] Train model
- [x] Record values of objective function (append to TSV)

### Misc.

- [x] Repeat for second optimization, taking a similar approach to the "identify associations" task (see [`move.tasks.identify_associations`](https://github.com/RasmussenLab/MOVE/blob/2541fcce531013b3506e0fa08f26d4e8f41442d4/src/move/tasks/identify_associations.py#L53)), i.e., detect type of task and change the value of the objective function (from accuracy to stabillity).
- [x] Add to `move.__main__`.
- [x] Re-format tutorial files for `random.small` dataset.

## Open Questions

- Is the best set of hyperparameters automatically selected based on the objective function value (e.g., reconstruction accuracy)?
    -  If so, I would suggest we also implement the Optuna plugin with a smarter sampler than greedy grid search.
    - If not, then I suggest just saving the results, and then providing some visualization functions so the users can decide on their hyperparameter set.
- [x] #22 

## Resources:

- Hydra's documentation on [multirun](https://hydra.cc/docs/1.2/tutorials/basic/running_your_app/multi-run/) and [sweepers](https://hydra.cc/docs/1.2/plugins/optuna_sweeper/)
- [Example Hydra app](https://github.com/ri-heme/example-hydra-app) I wrote with multirun examples
